Src/Support/Google.Apis.Auth.Tests/OAuth2/ServiceAccountCredentialTests.cs

```diff
@@ -408,6 +408,46 @@ public async Task Jwt_WithUserAndScopes_AccessToken()
             Assert.Equal("a", token);
         }
 
+        [Fact]
+        public async Task AccessToken_ExceptionRetry_NoDefaultRetry()
+        {
+            var clock = new MockClock(new DateTime(2016, 1, 1, 0, 0, 0, DateTimeKind.Utc));
+            var handler = new ResponseExcetionOnFetchingTokenMessageHandler();
+            var initializer = new ServiceAccountCredential.Initializer("some-id")
+            {
+                // Let's force access token fetching
+                UseJwtAccessWithScopes = false,
+                Scopes = new[] { "scope1", "scope2" },
+                Clock = clock,
+                HttpClientFactory = new MockHttpClientFactory(handler)
+            }.FromPrivateKey(PrivateKey);
+            var cred = new ServiceAccountCredential(initializer);
+
+            await Assert.ThrowsAsync<HttpRequestException>(() => cred.GetAccessTokenForRequestAsync("uri0"));
+            Assert.Equal(1, handler.Calls);
+        }
+
+        [Fact]
+        public async Task AccessToken_ExceptionRetry_ConfigedRetry()
+        {
+            var clock = new MockClock(new DateTime(2016, 1, 1, 0, 0, 0, DateTimeKind.Utc));
+            var handler = new ResponseExcetionOnFetchingTokenMessageHandler();
+            var initializer = new ServiceAccountCredential.Initializer("some-id")
+            {
+                // Let's force access token fetching
+                UseJwtAccessWithScopes = false,
+                Scopes = new[] { "scope1", "scope2" },
+                Clock = clock,
+                HttpClientFactory = new MockHttpClientFactory(handler),
+                DefaultExponentialBackOffPolicy = ExponentialBackOffPolicy.Exception | ExponentialBackOffPolicy.UnsuccessfulResponse503
```

Comment:
It feels slightly odd that this is called DefaultExponentialBackoffProperty given that it's not the default. I realise this is existing code and there might be a good reason for it - it just feels a bit odd...

Reply:
Yes, I don't like this name either.

I think it's called that, because in principle, users can add their own `Google.Apis.Http.IConfigurableHttpClientInitializer` via `ServiceAccountCredential.Initializer.HttpClientInitializers` which in turn can add request or response interceptors. Response interceptors added like that would be called after the "default" ones which are added depending on how this `DefaultExponentialBackoffProperty` is set.